library/std/src/sys/windows/c/windows_sys.rs

```diff
@@ -3875,47 +3875,6 @@ pub const SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE: SYMBOLIC_LINK_FLAGS = 2u
 pub const SYMBOLIC_LINK_FLAG_DIRECTORY: SYMBOLIC_LINK_FLAGS = 1u32;
 pub const SYMLINK_FLAG_RELATIVE: u32 = 1u32;
 pub const SYNCHRONIZE: FILE_ACCESS_RIGHTS = 1048576u32;
-#[repr(C)]
-pub struct SYSTEM_INFO {
```

Comment:
Miri was relying on this type declaration for its `GetSystemInfo` shim, that's why we are getting ICEs now.

The alternative is to change Miri's `GetSystemInfo` to use hard-coded offsets for the two fields it cares about (`dwPageSize`, `dwNumberOfProcessors`).

Reply:
I think it's ok to keep the old definitions around even though they're no longer used by std. I don't think there's yet a pressing need to keep `windows_sys.rs` definitions strictly minimal,. Plus keeping them reduces churn should we decide to use them for something else in the future.

Reply:
I can also disentangle Miri from that definition, if you'd prefer us not to rely on it. That can happen asynchronously with this PR, so eventually when the Miri patch lands you can do the cleanup.

Reply:
I mean, it might be better overall if miri wasn't entangled with the std this way but I don't feel particularly strongly about it.

Reply:
On Unix we rely on `libc` for tons of constants and types. That's just a lot more maintainable than hard-coding them all in the Miri sources. For Windows sadly there is no `libc`-equivalent so we can only use the few things we can find in std...

Reply:
Can you not use the [`windows-sys`](https://github.com/microsoft/windows-rs/blob/master/crates/libs/sys/src/lib.rs) crate directly? The definitions in std are essentially just a vendored version of it.

Reply:
We need a crate that's in the sysroot. (Miri might be running a Windows target on a Linux host.) So... we can do that once std depends on it. ;)